library/cwm/src/lib/cwm/pager.rb

```diff
@@ -96,30 +74,30 @@ def switch_page(page_id)
     abstract_method :mark_page
 
     # The contents will probably include a *selector*, such as {Tabs}
-    # or {Tree} and a {ReplacePoint} where {Page}s will appear.
+    # or {Tree} and a {replace_point} where {Page}s will appear.
     # @return [WidgetTerm]
     abstract_method :contents
 
-    # gets id of initial page
-    # This default implementation returns first page passed to constructor
-    def initial_page_id
+    # gets initial page
+    # This default page which return true for method initial otherwise first page passed
+    # to constructor
+    def initial_page
       initial = @pages.find(&:initial)
 
-      (initial || @pages.first).widget_id
+      initial || @pages.first
     end
 
-    def page_for_id(id)
-      @pages.find { |t| t.widget_id == id }
+    # gets id of initial page
+    def initial_page_id
```

Comment:
Let's drop this and let the only caller use initial_page.widget_id (or even compare pages instead of their ids, see that caller)